csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -595,23 +609,31 @@ private void ResolveConflicts(IEnumerable<string> frameworkPaths)
         /// <param name="projectFile">The project file making the reference.</param>
         private void UnresolvedReference(string id, string projectFile) => unresolvedReferences[id] = projectFile;
 
-        /// <summary>
-        /// Reads all the source files and references from the given list of projects.
-        /// </summary>
-        /// <param name="projectFiles">The list of projects to analyse.</param>
-        private void AnalyseProjectFiles(IEnumerable<FileInfo> projectFiles)
+        private void AnalyseSolutions(IEnumerable<string> solutions)
         {
-            foreach (var proj in projectFiles)
+            Parallel.ForEach(solutions, new ParallelOptions { MaxDegreeOfParallelism = options.Threads }, solutionFile =>
             {
-                AnalyseProject(proj);
-            }
+                try
+                {
+                    var sln = new SolutionFile(solutionFile);
+                    progressMonitor.LogInfo($"Analyzing {solutionFile}...");
+                    foreach (var proj in sln.Projects.Select(p => new FileInfo(p)))
```

Comment:
It looks like the filter `.Where(p => p.Exists)` was removed as a part of the re-factor. Is that intentional?

Reply:
`AnalyseProject` checks and checked for the existence of the files. Now if they don't exist, we'll log a message.